cni/pkg/nodeagent/server.go

```diff
@@ -66,6 +67,22 @@ type Server struct {
 	cniServerStopFunc func()
 }
 
+// runWithHostNs executes the provided function within the host network namespace
+func runWithHostNs(f func() error) error {
+	if f == nil {
+		return nil
+	}
+	hostNs := "/host/proc/1/ns/net"
+	if _, err := os.Stat(hostNs); os.IsNotExist(err) {
+		log.Warnf("Path %s does not exist, running function without switching to host network namespace\n", hostNs)
```

Comment:
Hmm, this seems potentionally dangerous. We might want to fail hard. Otherwise we will not configure the network right.

Plausibly at the startup, we can open the file descriptor and store it and pass it around. Then we have 1 point of failure at startup vs on every operation? I guess the `setns` can fail at anytime though

Reply:
Removed the check, as there was no good reason for this scenario to occurr, unless it is a unit test.
Now the hostNS Fd is created at boot and carried over to the netserver (which has the duty of closing the fd when stopping).
During boot I'm also checking if hostNS == currentNS by comparing the inodes, if they are the same then I will disable the network switching logic and the old behavior will be used instead